modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcTransactionTest.java

```diff
@@ -240,6 +243,37 @@ public void testRollbackInAutoCommitMode() throws SQLException  {
         }
     }
 
+    /**
+     * Ensure that explicit transaction can not be used, after it encounters an error.
+     */
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-20534")
+    @ParameterizedTest
+    @CsvSource({
+            // dml or not | SQL statement
+            "true,insert into TEST (ID) values (2)",
+            "false,SELECT * FROM test",
+    })
+    public void testOperationsFailsWhenTransactionEncoutersAnError(boolean dml, String sqlStmt) throws SQLException {
+        try (Connection conn = DriverManager.getConnection(URL)) {
+            conn.setAutoCommit(false);
+
+            try (Statement stmt = conn.createStatement()) {
+                assertThrowsSqlException(SQLException.class, () -> stmt.executeQuery("SELECT 1/0").next());
```

Comment:
Let's use variant with expected message instead of expected base SQLException class.

Reply:
Fixed.